daxa/process/erosita/clean.py

```diff
@@ -0,0 +1,322 @@
+    
+    # Converting to the right unit                              
+    else:
+        max_threshold = max_threshold.to('sb_rate')
+    
+    # Checking user's choice for the mask_iter parameter
+    if not isinstance(mask_iter, int):
+        raise TypeError("The mask_iter argument must be an integer.")
+
+    # Converting parameters from astropy units into a type the command line will accept
+    pimin = int(pimin.value)
+    pimax = int(pimax.value)
+    mask_pimin = int(mask_pimin.value)
+    mask_pimax = int(mask_pimax.value)
+    timebin = float(timebin.value)
+    source_size = float(source_size.value)
+    threshold = threshold.value
+    max_threshold = max_threshold.value
+
+    # These parameters we want DAXA to have control over, not the user
+    gridsize = 18   # Sections of the image a source detection is run over to determine a dynamic threshold
+    fov_radius = 30 # Not sure about this parameter yet
+    xmin = -108000  # These are for making the image
+    xmax = 108000
+    ymin = -108000
+    ymax = 108000
+    write_thresholdimg = 'yes'
+    write_mask = 'yes'
+    write_lightcurve = 'yes' 
+
+    # Sets up storage dictionaries for bash commands, final file paths (to check they exist at the end), and any
+    #  extra information that might be useful to provide to the next step in the generation process
+    miss_cmds = {}
+    miss_final_paths = {}
+    miss_extras = {}
+
+    # Just grabs the eROSITA missions, we already know there will be at least one because otherwise _esass_process_setup
+    #  would have thrown an error
+    erosita_miss = [mission for mission in obs_archive if mission.name in ALLOWED_EROSITA_MISSIONS]
+    # We are iterating through XMM missions (options could include xmm_pointed and xmm_slew for instance).
+    for miss in erosita_miss:
+        # Sets up the top level keys (mission name) in our storage dictionaries
+        miss_cmds[miss.name] = {}
+        miss_final_paths[miss.name] = {}
+        miss_extras[miss.name] = {}
+
+        # This method will fetch the valid data (ObsID, Instruments) that can be processed
+        all_obs_info = obs_archive.get_obs_to_process(miss.name)
+            
+        # Checking that any valid observations are left after the get_obs_to_process function is run
+        if len(all_obs_info) == 0:
+            raise FileNotFoundError("No valid observations have been found, so flaregti may not be run.")
+
+        # all_obs_info is a list of lists, where each list is of the format: [ObsID, Inst, 'usable'].
+        #   There is a new list for each instrument, but I just want to loop over the ObsID in the following bit of code, 
+        #   I also want to know all the instruments that the ObsID contains events for
+        #   So here I am just making a dictionary of the format: {ObsID: insts}
+        # Getting unique obs_ids in all_obs_info
+        obs_ids = list(set([all_obs_info_list[0] for all_obs_info_list in all_obs_info]))
```

Comment:
Yes, I remember!

So the 'all_obs_info' parameter is a lists of lists, where each list is of the form [ObsID, instrument, 'usable' ].  These lists are generated from the observation summaries which are defined in the archive.

In the case of XMM, you have the observation summaries also containing subexposure information, so when you define the all_obs_info parameter you have another search level: 
rel_m_obs = obs_archive.get_obs_to_process(miss.name, 'M1') + obs_archive.get_obs_to_process(miss.name, 'M2')
(all_obs_info is the equivalent to rel_m_obs)

So for the erosita case, I just needed to gather all the observations in a bit of a convoluted way, such as in line 251 of this pull request conversation. And then to find all the instruments associated with that observation, (for file naming purposes) I need to have a line like:
insts = ''.join([all_obs_info_list[1] for all_obs_info_list in all_obs_info if obs in all_obs_info_list])

So it was just these convoluted ways of extracting obs_ids and instrument information from the obs_archive.get_obs_to_process(miss.name) parameter that was a bit of a bodge, since it is more optimised for XMM style observations.

It still works fine, and isnt too slow though, so I think it is fine to leave for now?

Reply:
It is fine to leave for now, but please make an issue if you haven't already so we can remember to change this at some point. Then you can resolve this conversation!